src/core/queryCache.ts

```diff
@@ -80,6 +80,15 @@ export class QueryCache extends Subscribable<QueryCacheListener> {
     }
   }
 
+  reset(query: Query<any, any>): void {
+    const queryInMap = this.queriesMap[query.queryHash]
+
+    if (queryInMap) {
+      query.reset()
+      this.notify(query)
```

Comment:
The query cache will already be notified when calling `query.reset()`. But maybe we don't even need these methods on the query cache?